test/WebJobs.Script.Tests.Integration/Security/AuthenticationLevelHandlerTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Authentication;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Hosting;
+using Microsoft.WebJobs.Script.Tests;
+using Moq;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Integration.Security
+{
+    [Trait(TestTraits.Category, TestTraits.EndToEnd)]
+    [Trait(TestTraits.Group, nameof(AuthenticationLevelHandlerTests))]
+    public class AuthenticationLevelHandlerTests
+    // TODO: This should be a more general E2E test suite - maybe for SWA? Or combine with storage/secret E3E tests?
+    {
+        public static IEnumerable<object[]> GetData()
+        {
+            yield return new object[] { @"TestScripts\CSharp", RpcWorkerConstants.DotNetLanguageWorkerName,
+                new Dictionary<string, string>() { ["AzureWebJobsStorage"] = string.Empty }};
+
+            // test when AzureWebJobsSecretStorageType=None
+            // test when AzureWebJobsSecretStorageType= something else
+            // test when AzureWebJobsSecretStorageType is not set
+
+            // test for Node scenarios too
+        }
+
+        [Theory]
+        [MemberData(nameof(GetData))]
+        public async Task NoStorage_ReturnsExpected(string scriptRoot, string runtime, IDictionary<string, string> config)
```

Comment:
We need tests for other SWA scenarios. I sketched some out in https://github.com/Azure/azure-functions-host/blob/eb5e2e2f003634d42adf01ade43a1c5e7477850d/test/WebJobs.Script.Tests.Integration/ScriptHostEndToEnd/SWAEndToEndTests.cs#L34